src/os/bluestore/KernelDevice.cc

```diff
@@ -444,17 +455,20 @@ int KernelDevice::aio_zero(
   assert(len > 0);
   assert(off < size);
   assert(off + len <= size);
-
-  bufferlist bl;
-  while (len > 0) {
-    bufferlist t;
-    t.append(zeros, 0, MIN(zeros.length(), len));
-    len -= t.length();
-    bl.claim_append(t);
+  if (discard_zeroes_data) {
+    return block_device_discard(fd_direct, off, len);
```

Comment:
@yuyuyu101 . From the user space, write zero mean later read data is zero. So the discard and write-zero is the same. Further, discard for device is good.
I agree add aio_discard because we need this when free blocks we should discard those.